tests/phpunit/tests/html-api/wpHtmlTagProcessor-token-scanning.php

```diff
@@ -760,6 +760,126 @@ public function test_basic_assertion_funky_comment() {
 		);
 	}
 
+	/**
+	 * Ensures that various funky comments are properly parsed.
+	 *
+	 * @ticket 60170
+	 *
+	 * @since 6.6.6
+	 *
+	 * @covers WP_HTML_Tag_Processor::next_token
+	 *
+	 * @dataProvider data_various_funky_comments
+	 *
+	 * @param string $funky_comment_html HTML containing a funky comment.
+	 * @param string $modifiable_text    Expected modifiable text of first funky comment in HTML.
+	 */
+	public function test_various_funky_comments( $funky_comment_html, $modifiable_text ) {
+		$processor = new WP_HTML_Tag_Processor( $funky_comment_html );
+		while ( '#funky-comment' !== $processor->get_token_type() && $processor->next_token() ) {
+			continue;
+		}
+
+		$this->assertSame(
+			'#funky-comment',
+			$processor->get_token_type(),
+			'Failed to find the expected funky comment.'
+		);
+
+		$this->assertSame(
+			$modifiable_text,
+			$processor->get_modifiable_text(),
+			'Found the wrong modifiable text span inside a funky comment.'
+		);
+	}
+
+	/**
+	 * Data provider.
+	 *
+	 * @return array[].
+	 */
+	public static function data_various_funky_comments() {
+		return array(
+			'Space'             => array( '</ >', ' ' ),
+			'Short-bang'        => array( '</!>', '!' ),
+			'Markup Decl. Like' => array( '</?>', '?' ),
```

Comment:
This descriptive text seems less obvious than the others, I don't know what it means.

Reply:
I updated the wording to note a question mark. It was intended to indicate "looks like markdown declaration state" but that didn't need to be.